client/src/main/com/sinch/sdk/domains/conversation/adapters/MessagesService.java

```diff
@@ -0,0 +1,62 @@
+package com.sinch.sdk.domains.conversation.adapters;
+
+import com.sinch.sdk.core.exceptions.ApiException;
+import com.sinch.sdk.core.http.AuthManager;
+import com.sinch.sdk.core.http.HttpClient;
+import com.sinch.sdk.core.http.HttpMapper;
+import com.sinch.sdk.core.utils.databind.Mapper;
+import com.sinch.sdk.domains.conversation.adapters.messages.SendMessageRequestMapper;
+import com.sinch.sdk.domains.conversation.api.v1.MessagesApi;
+import com.sinch.sdk.domains.conversation.models.v1.messages.request.SendMessageRequest;
+import com.sinch.sdk.domains.conversation.models.v1.messages.request.SendMessageRequestImpl;
+import com.sinch.sdk.domains.conversation.models.v1.messages.response.SendMessageResponse;
+import com.sinch.sdk.domains.conversation.models.v1.response.ConversationMessage;
+import com.sinch.sdk.models.ConversationContext;
+import java.util.Map;
+
+public class MessagesService implements com.sinch.sdk.domains.conversation.MessagesService {
+
+  static {
+    LocalLazyInit.init();
+  }
+
+  private final String uriUUID;
+  private final MessagesApi api;
+
+  public MessagesService(
+      String uriUUID,
+      ConversationContext context,
+      HttpClient httpClient,
+      Map<String, AuthManager> authManagers) {
+    this.uriUUID = uriUUID;
+    this.api = new MessagesApi(httpClient, context.getServer(), authManagers, new HttpMapper());
+  }
+
+  protected MessagesApi getApi() {
+    return this.api;
+  }
+
+  public SendMessageResponse send(SendMessageRequest request) throws ApiException {
+    return getApi().messagesSendMessage(uriUUID, request);
+  }
+
+  public ConversationMessage get(String messageId) {
+    return getApi().messagesGetMessage(uriUUID, messageId, null);
+  }
+
+  static final class LocalLazyInit {
```

Comment:
This is to avoid to tie the `MessagesService` to core.
`core` is containing the mapper instance to sererialize/deserialize payloads.
Registering the SendMessageRequestMapper dedicated class (required by **Conversation domain only**) from core will lead to hard couple between core and conversation
The aim of core part it to be domain's agnostic: so no static relationship
To avoid this: the domain dedicated settings (extending mapper here) is involved to the domain itself.
Finally, it follow same logic as checking domain's related credentials : it is not a general core task but a dedicated domain's task